src/ceph-volume/plugin/cache/ceph_volume_cache/main.py

```diff
@@ -0,0 +1,207 @@
+        return disk.Size(kb=n)
+    return None
+
+
+# partition sizes in GB
+def _create_cache_lvs(vg_name, md_partition, data_partition):
+    md_partition_size = strToSize(disk.lsblk(md_partition)['SIZE'])
+    data_partition_size = strToSize(disk.lsblk(data_partition)['SIZE'])
+
+    if not md_partition_size >= disk.Size(gb=2):
+        print('Metadata partition is too small')
+        return
+    if not data_partition_size >= disk.Size(gb=2):
+        print('Data partition is too small')
+        return
+
+    # ceph-volume creates volumes with extent size = 1GB
+    # when a new lv is created, one extent needs to be used by LVM itself
+    md_lv_size = md_partition_size - disk.Size(gb=1)
+    data_lv_size = data_partition_size - disk.Size(gb=1)
+
+    # TODO: update tags
+    cache_md_lv = api.create_lv('cache_metadata', vg_name, extents=None,
+        size=str(md_lv_size._b) + 'B', tags=None, uuid_name=False, pv=md_partition)
+    cache_md_lv.set_tag('ceph.cachetype', 'metadata')
+    cache_md_lv.set_tag('ceph.partition', md_partition)
+    cache_data_lv = api.create_lv('cache_data', vg_name, extents=None,
+        size=str(data_lv_size._b) + 'B', tags=None, uuid_name=False, pv=data_partition)
+    cache_data_lv.set_tag('ceph.cachetype', 'data')
+    cache_data_lv.set_tag('ceph.partition', data_partition)
+    # TODO: this tags is probably not needed, we can get it using '_cmeta'
+    cache_md_lv.set_tag('ceph.metadata', cache_md_lv.name)
+
+    return cache_md_lv, cache_data_lv
+
+
+def _create_lvmcache(vg_name, osd_lv_name, cache_metadata_lv_name, cache_data_lv_name):
+    osd_lv = api.get_lv(lv_name=osd_lv_name, vg_name=vg_name)
+    # TODO: test that cache data is greater than metadata
+    api.create_lvmcache_pool(vg_name, cache_data_lv_name, cache_metadata_lv_name)
+    api.create_lvmcache(vg_name, cache_data_lv_name, osd_lv_name)
+
+    osd_lv.set_tag('ceph.cache_lv', cache_data_lv_name)
+    api.set_lvmcache_caching_mode('writeback', vg_name, osd_lv_name)
+
+    _cache_data_lv_name = '[' + cache_data_lv_name + ']'
+    cache_lv = api.get_lv(lv_name=_cache_data_lv_name, vg_name=vg_name)
+    return cache_lv
+
+
+def add_lvmcache(vgname, osd_lv_name, md_partition, cache_data_partition):
+    """
+    High-level function to be called. Expects the user or orchestrator to have
+    partitioned the disk used for caching.
+    """
+    # TODO add pvcreate step?
+    vg = api.get_vg(vg_name=vgname)
+    # TODO don't fail if the LVs are already part of the vg
+    api.extend_vg(vg, [md_partition, cache_data_partition])
+    cache_md_lv, cache_data_lv = _create_cache_lvs(vg.name, md_partition, cache_data_partition)
+    cachelv = _create_lvmcache(vg.name, osd_lv_name, cache_md_lv.name, cache_data_lv.name)
+
+    return cachelv
+
+
+def rm_lvmcache(vgname, osd_lv_name):
+    osd_lv = api.get_lv(lv_name=osd_lv_name, vg_name=vgname)
+    if osd_lv.tags['ceph.cache_lv'] is None or len(osd_lv.tags['ceph.cache_lv']) == 0:
```

Comment:
Since 'is None' and '== 0' are equivalent conditions in this case, the if can be written as:
```
if not osd_lv.tags['ceph.cache_lv']:
```